typescript/cli/src/deploy/core.ts

```diff
@@ -323,8 +315,18 @@ async function executeDeploy({
     multisigConfigs ?? defaultMultisigConfigs, // TODO: fix https://github.com/hyperlane-xyz/issues/issues/773
   );
   const coreContracts = await coreDeployer.deploy(coreConfigs);
+
+  // 4.5 recover the toplevel ISM address
+  const isms: HyperlaneAddressesMap<any> = {};
+  for (const chain of chains) {
+    isms[chain] = {
+      interchainSecurityModule:
+        coreDeployer.cachedAddresses[chain].interchainSecurityModule,
+    };
+  }
```

Comment:
we could also fetch this from the mailbox?

Reply:
where is this used?

Reply:
fetching from mailbox is more repetitive bc we're already getting the up-to-date ism from coreDeployer. cachedAddresses in both cases where we deploy and/or recover

Reply:
it's used for TestRecipient